### ISSUE TITLE
Remove usages of deprecated org.apache.sling.commons.json

### ISSUE THREAD

**Issue by rombert** [Adobe-Consulting-Services/acs-aem-commons#3256]
## Required Information

* [x] AEM Version, including Service Packs, Cumulative Fix Packs, etc: Not applicable
* [x] ACS AEM Commons Version: laster master build
* [x] Reproducible on Latest? yes

### Expected Behavior

The `acs-aem-commons` bundle should not import packages provided by `org.apache.sling.commons.json`. The bundle providing those packages has been deprecated  since March 2017 without any replacement.

### Actual Behavior

I see #943 has been closed but I still acs-aem-commons still has imports for commons.json

### Steps to Reproduce

```
$ mvn clean package
$ bnd grep --imports org.apache.sling.commons.json bundle/target/acs-aem-commons-bundle-6.3.9-SNAPSHOT.jar 
bundle/target/acs-aem-commons-bundle-6.3.9-SNAPSHOT.jar :       Import-Package ...[org.apache.sling.commons.json]...
```

### Links

- https://experienceleague.adobe.com/docs/experience-manager-cloud-service/content/release-notes/deprecated-removed-features.html?lang=en
- https://lists.apache.org/thread/ctsqmtojokrhz249tzhdlq11x5mxl3xs
- https://issues.apache.org/jira/browse/SLING-6536

**Comment by csaboka** [comment]
The only places this package is referenced (aside from unit tests) are two implementations of [PageInfoProvider](https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/com/day/cq/wcm/api/PageInfoProvider.html). That interface references org.apache.sling.commons.json.JSONObject in the signature of its single method, so it's impossible to implement it without using deprecated types. I'm not even sure how Adobe is planning to drop this package from the API when PageInfoProvider isn't deprecated itself, and as far as I know, there is no replacement that's free from this dependency. All I can say is that the code base is in a good shape to get rid of this dependency once a supported alternative is made available.

**Comment by cziegeler** [comment]
@kwin  I am thinking of creating a new interface which does not use JSON at all, it will have a method that returns a map of String, Object - I believe inside AEM the JSON is converted to such a map anyway. If I am overlooking something, please let me know

**Comment by kwin** [comment]
>  it will have a method that returns a map

AFAIK we are only using https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/com/day/cq/wcm/api/PageInfoProvider.html where org.apache.sling.commons.json.* is not used as return type but argument type. Are you talking about a different interface, @cziegeler?

**Comment by cziegeler** [comment]
@kwin yes, it is exactly that interface - it has a update method which gets a JSONObject and the PageInfoProviders are supposed to add properties to that object.
Instead of passing a map into an update method, we could use a map as a return type, creating a slightly nicer interface

**Comment by kwin** [comment]
Got it, so something like `Map<String, Object> getInfo(SlingHttpServletRequest, Resource)`? Sounds good to me. I am just not sure about PageProvider interdependencies, i.e. one provider leveraging an attribute contributed by a preceding provider...

**Comment by stefanseifert** [comment]
@cziegeler when you are working on getting rid of `org.apache.sling.commons.json` in the AEM API, please keep in mind there are other parts of the API that are affected as well.
for example, https://wcm.io/handler/commons/ depends on [DropTarget](https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/com/day/cq/wcm/api/components/DropTarget.html) which extends [JSONItem](https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/com/day/cq/commons/JSONItem.html) and references [JSONWriter](https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/org/apache/sling/commons/json/io/JSONWriter.html).
we are aware of this since years, but as there was no fix or workaround in the AEM API, we could not get rid of it.

**Comment by kwin** [comment]
Also we we would appreciate a common API for AEM 6.5 LTS to not be forced to use different code in libraries supporting both AEM 6.5/Cloud.

**Comment by stefanseifert** [comment]
> Also we we would appreciate a common API for AEM 6.5 LTS to not be forced to use different code in libraries supporting both AEM 6.5/Cloud.

yes, fully agree, i forget to mention this. backporting this is very important (maybe even backport it to AEM 6.5.x - we are still supporting it for the wcm.io libraries). also make sure to keep the OSGi package versions in sync...

**Comment by henrykuijpers** [comment]
Is this already on the roadmap @cziegeler ?
